### PR TITLE
Delegates

### PR DESCRIPTION
**Pull request by smartcontrart**

### CODE REVIEW

**Review by smartcontrart — COMMENTED** [review]
I had a bug with the commit message and the wrong one went through.
This PR is to remove an unused folder and revert the custom abis we had to match develop/main

File: source/delegate/LICENSE [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Copyright 2023 AirSwap
```
Comment (dmosites) [inline source/delegate/LICENSE:1]: 2024

File: source/delegate/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Pool
```
Comment (dmosites) [inline source/delegate/README.md:1]: Delegate

File: source/delegate/contracts/Delegate.sol [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    address _senderToken,
+    address _signerToken
+  ) external {
+    if (authorized[_senderWallet] != address(0)) {
+      if (authorized[_senderWallet] != msg.sender) revert SenderInvalid();
+    } else {
+      if (_senderWallet != msg.sender) revert SenderInvalid();
+    }
+    Rule storage rule = rules[_senderWallet][_senderToken][_signerToken];
+    rule.senderAmount = 0;
```
Comment (dmosites) [inline source/delegate/contracts/Delegate.sol:87]: Would we gain anything by zeroing out the other fields?

File: source/delegate/contracts/Delegate.sol [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    Rule storage rule = rules[_senderWallet][_senderToken][_signerToken];
+
+    if (
+      _senderAmount > (_signerAmount * rule.senderAmount) / rule.signerAmount
+    ) {
+      revert InsufficientSignerAmount();
+    }
+
+    if (rule.senderAmount < _senderAmount) {
+      revert InsufficientDelegateAllowance();
```
Comment (dmosites) [inline source/delegate/contracts/Delegate.sol:113]: "InvalidSenderAmount" to match the above.

File: source/delegate/contracts/Delegate.sol [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+    SafeTransferLib.safeTransfer(_signerToken, _senderWallet, _signerAmount);
+
+    rules[_senderWallet][_senderToken][_signerToken]
+      .senderAmount -= _senderAmount;
+    emit DelegateSwap(_nonce, _signerWallet);
+  }
+
+  /**
+   * @notice Authorize a signatory
```
Comment (dmosites) [inline source/delegate/contracts/Delegate.sol:146]: "Authorize a wallet to manage rules"

File: source/delegate/contracts/Delegate.sol [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+   * @dev Emits an Authorize event
+   */
+  function authorize(address _manager) external {
+    if (_manager == address(0)) revert ManagerInvalid();
+    authorized[msg.sender] = _manager;
+    emit Authorize(_manager, msg.sender);
+  }
+
+  /**
+   * @notice Revoke the signatory
```
Comment (dmosites) [inline source/delegate/contracts/Delegate.sol:157]: "Revoke a manager"

File: source/delegate/package.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@airswap/delegate",
+  "version": "4.1.4",
```
Comment (dmosites) [inline source/delegate/package.json:3]: `4.3.0-beta.0`

File: source/delegate/package.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  "description": "AirSwap: Trade intent delegates",
```
Comment (dmosites) [inline source/delegate/package.json:4]: "AirSwap: Delegate On-chain Trading Rules"

File: source/delegate/contracts/Delegate.sol [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+  mapping(address => mapping(address => mapping(address => Rule))) public rules;
+
+  // Mapping of signer to authorized signatory
+  mapping(address => address) public authorized;
+
+  /**
+   * @notice Contract Constructor
+   */
+  constructor(ISwapERC20 _swapERC20) {
+    swapERC20 = _swapERC20;
```
Comment (dmosites) [inline source/delegate/contracts/Delegate.sol:30]: Probably also want a setter for swapERC20 similar to Wrapper.

File: source/delegate/contracts/Delegate.sol (hunk 3 above)
Comment (smartcontrart) [reply source/delegate/contracts/Delegate.sol:87]: I originally didn't do it to save on gas. But the additional security might be worth the gas (which won't be much)

File: source/delegate/README.md (hunk 2 above)
Comment (smartcontrart) [reply source/delegate/README.md:1]: Done

File: source/delegate/contracts/Delegate.sol (hunk 5 above)
Comment (smartcontrart) [reply source/delegate/contracts/Delegate.sol:146]: Done

File: source/delegate/package.json (hunk 7 above)
Comment (smartcontrart) [reply source/delegate/package.json:3]: Done

File: source/delegate/contracts/Delegate.sol (hunk 9 above)
Comment (smartcontrart) [reply source/delegate/contracts/Delegate.sol:30]: Done

File: source/delegate/contracts/Delegate.sol (hunk 3 above)
Comment (dmosites) [reply source/delegate/contracts/Delegate.sol:90]: How about just something like `delete rules[_senderWallet][_senderToken][_signerToken];`

Comment (smartcontrart) [reply source/delegate/contracts/Delegate.sol:90]: Absolutely... No idea why I wrote that horror...

**Review by dmosites — APPROVED** [review]

**Pull request merged**